jwst/datamodels/model_base.py

```diff
@@ -335,6 +335,27 @@ def validate(self):
                               self._pass_invalid_values,
                               self._strict_validation)
 
+    def validate_required_fields(self):
+        """
+        Walk the schema and make sure all required fields are
+        in the model
+        """
+        def callback(schema, path, combiner, ctx, recurse):
+            if 'fits_required' not in schema:
+                return
+
+            node = ctx
```

Comment:
Curiosity question: After this loop, there are only three possible outcomes: node is ctx, node is None, or node is the last attribute. Any other attributes in path are ignored. Is this intended?

Reply:
The path is what you get after you split the schema field you are looking for, For example, 'meta.program.title' becomes ['meta', 'program', 'title']. The reason for the test for None is that one of the intermediates may be missing. In this example, there may be 'meta', but no 'meta.program', in which case there is also no 'meta.program.title'. Checking for None prevents the error of trying to find 'title' as the attribute of None and also returns the correct result from the loop, which is None.

Reply:
A comment to this effect may be in order, otherwise 👍